src/SVGSkin.js

```diff
@@ -91,6 +119,13 @@ class SVGSkin extends Skin {
                 this._texture = twgl.createTexture(gl, textureOptions);
                 this._silhouette.update(this._svgRenderer.canvas);
             }
+
+            const maxDimension = Math.max(this._svgRenderer.canvas.width, this._svgRenderer.canvas.height);
+            let testScale = 2;
+            for (testScale; maxDimension * testScale <= MAX_TEXTURE_DIMENSION; testScale *= 2) {
+                this._maxTextureScale = testScale;
```

Comment:
This block might benefit from a comment, at first it wasn't clear what was going on here. Is my understanding right: it looks like we are precomputing the max texture scale after rendering the texture the first time, when it was rendered presumably at a scale of 1. This was surprising to me at first because we have a very similar loop in `getTexture`, but it is running on scale instead of dimensions.

Reply:
Yeah, I'm not very happy with the way this is set up. I can certainly add a comment but I'd like to find a way to simplify this code now that the basics are working.